src/test/base64_tests.cpp

```diff
@@ -37,10 +37,22 @@ BOOST_AUTO_TEST_CASE(base64_testvectors)
     }
 
     // Decoding strings with embedded NUL characters should fail
-    BOOST_CHECK(!DecodeBase64("invalid\0"s));
-    BOOST_CHECK(DecodeBase64("nQB/pZw="s));
-    BOOST_CHECK(!DecodeBase64("nQB/pZw=\0invalid"s));
-    BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid\0"s));
+    BOOST_CHECK(!DecodeBase64("invalid\0"s)); // correct size, invalid due to \0
+    BOOST_CHECK( DecodeBase64("nQB/pZw="s)); // valid
+    BOOST_CHECK(!DecodeBase64("nQB/pZw=\0invalid"s)); // correct size, invalid due to \0
+    BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid\0"s)); // invalid size
```

Comment:
I forgot about this comment, thanks for checking, both tests on 42 and 43 were invalid because of the non-trailing `=`, not the `\0` - [updated them](https://github.com/bitcoin/bitcoin/compare/6fd185c035c1cc4dd961cf14a2087e97fb069440..37e5fcfd4fad2f2daed041ddfd60b0c28e5da6b6)

Reply:
They are invalid, but still worth testing regardless of why precisely the current implementation errors out?

```C++
BOOST_CHECK(!DecodeBase64("nQB/pZw=\0invalid"); // padding and NUL followed by invalid data
BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid\0"); // padding followed by invalid data and NUL
```

Reply:
What's the advantage of the comment stating exactly what the code already states?
We're already testing `\0` and non-trailing `=`, what else do these ones add?

Reply:
> What's the advantage of the comment stating exactly what the code already states? 

Valid point. Better to describe what is wrong according to the Base64 standard (+ `\0` I guess).

> We're already testing `\0` and non-trailing `=`, what else do these ones add?

They cover slightly different cases, mixing of `\0` and `=`. They may not test a unique error-handling path with the current implementation, but if they can uncover issues with a potential future implementation I think it was a mistake to remove them (@maflcko was just stating the comments were invalid as I understand it).

Reply:
Done, thanks